### PR TITLE
Tranks implementation

### PR DESCRIPTION
**Pull request by NathanZiri**
Added 2 tracks to the game with AI following the track path

### CODE REVIEW

**Review by Aether700 — CHANGES_REQUESTED** [review]
The mountain track does not have a target at the end of it nor does it have a bird spot where the racers can get attacked by birds. 

For the cayon, the rings in the cayon do not fit the esthetic of the game, would it be possible to use arrows like in mario kart to guide the racers instead of the rings?

Also, in the cayon track nothing stops the player from just flying above the cayon, cut directly accross straight to the target. Can you please look into making sure that is not possible.

Finally, the branch is not up to date with main. Can you please update the branch to pull from main. To double check that everything works well make sure that the racers get destroyed when they hit the target at the end of the track.

You will also need to set up the game over screen once that is all done. if you need help with that part let me know

**Comment by NathanZiri** [comment]
I will discuss with yuwen to address these issues and will pull update the branch from main to ensure it is more upto dare when being re pushed.

File: Assets/Scripts/PositionTracker.cs [hunk 1]
```diff
@@ -69,6 +73,7 @@ private void FinishRace(Bullseye bullseyeObj, Vector3 bullseyeHitPoint)
         score.SetAccuracyScore(bullseyeObj.GetScore(bullseyeHitPoint));
 
         // Compute time score here
+        score.SetTimeScore((int)(Time.time - startTime));
```
Comment (Aether700) [inline Assets/Scripts/PositionTracker.cs]: Would it be possible to have the time score be a "positive" number? as in the greater the time score the greater the final score?

For example we could have the first racer to finish always having a set maximum score and then decrease the score of the other racers based on the time they take compared to that initial time?

**Review by Aether700 — CHANGES_REQUESTED** [review]
In the mountain scene the bullseye is not at the end of the track. Please either update the track to end at the bullseye target or move the bullseye target at the end of the track. Also, most of the environment of the mountain scene has no colliders. Please add colliders to the scene and make sure that the AI avoids those colliders by using the correct layers of setting the layer mask of the AI correctly (I can help with that if necessary). Additionally the mountain track still has no birds on it. Please add some.

For the cayon scene, the bullseye is slightly off of the end of the track. you can easily modify that by having the bullseye target transform be the final position of the race track (I can help with that if you need to).

I also noticed that you are using mesh colliders for a lot of things when we can easily get away with other simpler colliders which run faster. For example. The plane collider in the mountain scene and the glass ceilling collider in the cayon scene both use mesh colliders which is really expensive when they could easily use box colliders. The same goes for the launch pad which would use a box collider.

**Review by Aether700 — COMMENTED** [review]
Good job. The Canyon in the canyon scene doesn't have any colliders on it. you can probably just do a mesh collider on that one given how complex it is.

For the mountain scene the bird area is kind of short. you should make it bigger. There is also a collider on a plane under the landing pad in the mountain scene which has a mesh collider on it. Given where in the scene it is you could probably just remove the collider but if you think it should keep it's collider use a box collider instead.

**Comment by Aether700** [comment]
There are merge conflicts preventing from properly pulling the changes. Can you please fix them?

**Review by Aether700 — APPROVED** [review]
Great work

**Pull request merged**